torch/_inductor/ir.py

```diff
@@ -190,6 +190,42 @@ def ir_node_to_tensor(x, guard_shape=True):
     return t
 
 
+class OptionalAttr:
+    def __repr__(self):
+        return self.name
+
+
+class OptionalString(OptionalAttr):
+    def __init__(self):
+        self.name = "optional_string"
+
+
+class OptionalList(OptionalAttr):
+    def __init__(self):
+        self.name = "optional_list"
+
+
+class OptionalScalar(OptionalAttr):
+    def __init__(self):
+        self.name = "optional_scalar"
```

Comment:
maybe it is simpler to make these as enums?

Reply:
Hi @jgong5 these classes inherit the `OptionalAttr` since I need to define the `__repr__` func of these classes and use the `__repr__` in the codegen. The codegen for these optional will go to the final `else` in the below code:

https://github.com/pytorch/pytorch/blob/c03555a303c645303b0e0c52336c30e5f0a7f21f/torch/_inductor/codegen/wrapper.py#L1067-L1083


If I use an enum like below:
```
class OptionalAttrEnum(Enum):
    optional_string = auto()
    optional_list = auto()
    optional_scalar = auto()
```
I fail to define the `__repr__` of `OptionalAttrEnum.optional_string` but only the `__repr__` of the `OptionalAttrEnum` in this way.

Option 1: I will be keeping the current implementation.
Option 2: I can use `enum` but need to add another `elif isinstance(val, OptionalAttrEnum):` in the `def val_to_str(self, val):` function to define the codegen for this class.

May I know which option is preferred?

Reply:
OK, option 1 LGTM then.